src/Sarif.Multitool/Rules/SARIF2004.OptimizeFileSize.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+using Microsoft.Json.Pointer;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class OptimizeFileSize : SarifValidationSkimmerBase
+    {
+        /// <summary>
+        /// SARIF2004
+        /// </summary>
+        public override string Id => RuleId.OptimizeFileSize;
+
+        /// <summary>
+        /// Placeholder_SARIF2004_OptimizeFileSize_FullDescription_Text
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF2004_OptimizeFileSize_FullDescription_Text };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+                    nameof(RuleResources.SARIF2004_OptimizeFileSize_Warning_EliminateLocationOnlyArtifacts_Text)
+                };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+        protected override void Analyze(Run run, string runPointer)
+        {
+            // We only verify first item in the results and artifacts array,
+            // since tools will typically generate similar nodes.
+            // This approach may cause false negatives.
+            ArtifactLocation firstLocationInArtifactsArray = run.Artifacts[0].Location;
+            ArtifactLocation firstlocationInResults = run.Results[0].Locations[0].PhysicalLocation.ArtifactLocation;
+
+            if (!ArrayBasedLocationHasAdditionalInfo(firstLocationInArtifactsArray, firstlocationInResults))
+            {
+                // {0}: Placeholder_SARIF2004_OptimizeFileSize_Warning_EliminateLocationOnlyArtifacts_Text
+                LogResult(
+                    runPointer.AtProperty(SarifPropertyName.Artifacts),
+                    nameof(RuleResources.SARIF2004_OptimizeFileSize_Warning_EliminateLocationOnlyArtifacts_Text));
+            }
+        }
+
+        private bool ArrayBasedLocationHasAdditionalInfo(ArtifactLocation arrayBasedLocation, ArtifactLocation resultBasedLocation)
```

Comment:
>ArrayBasedLocationHasAdditionalInfo [](start = 21, length = 35)

if the logic is indeed what we want to verify, then the name of this sub-rule needs to be changed. we are trying to verify if the array based location has any additonal info vs what result has.

Reply:
Not quite: Let's discuss the logic offline, and I'll craft a pseudo-SARIF sample to illustrate.

---
In reply to: [445260360](https://github.com/microsoft/sarif-sdk/pull/1939#discussion_r445260360) [](ancestors = 445260360)